source/docs/software/hardware-apis/sensors/gyros-software.rst

```diff
@@ -323,6 +399,41 @@ The following example shows how to stabilize heading using a simple P loop close
             drive.TankDrive(.5 + kP * error, .5 - kP * error);
         }
 
+    .. code-block:: python
+
+        from wpilib import Spark
+        from wpilib import MotorControllerGroup
+        from wpilib.drive import DifferentialDrive
+
+        # Use gyro declaration from above here
+
+        # The gain for a simple P loop
+        kP = 1
+
+        # Initialize motor controllers and drive
+        left1 = Spark(0)
+        left2 = Spark(1)
+        right1 = Spark(2)
+        right2 = Spark(3)
+
+        leftMotors = MotorControllerGroup(left1, left2)
+        rightMotors = MotorControllerGroup(right1, right2)
+
+        drive = DifferentialDrive(leftMotors, rightMotors)
+
+        def robotInit(self):
+            rightMotors.setInverted(true)
+
+        def autonomousInit(self):
+            # Set setpoint to current heading at start of auto
+            heading = gyro.getAngle()
```

Comment:
heading is only valid in this scope, `self.heading` needs to be used so the other functions can access it.